src/vs/workbench/contrib/codeEditor/browser/editorLineNumberMenu.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Separator } from 'vs/base/common/actions';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { ICodeEditor, MouseTargetType } from 'vs/editor/browser/editorBrowser';
+import { registerEditorContribution, EditorContributionInstantiation } from 'vs/editor/browser/editorExtensions';
+import { IEditorContribution } from 'vs/editor/common/editorCommon';
+import { IMenuService, MenuId } from 'vs/platform/actions/common/actions';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { IContextMenuService } from 'vs/platform/contextview/browser/contextView';
+import { IBreakpointEditorContribution, BREAKPOINT_EDITOR_CONTRIBUTION_ID } from 'vs/workbench/contrib/debug/common/debug';
+
+class EditorLineNumberContextMenu extends Disposable implements IEditorContribution {
+	private readonly menu = this._register(this.menuService.createMenu(MenuId.EditorLineNumberContext, this.contextKeyService));
```

Comment:
For a context menu it is actually favourable to create and dispose it on demand, e.g. `this.editor.onContextMenu`, because menu comes with the cost of being context key listeners - of which there are plenty and which fire a lot